static/downloadables/parse_associations.py

```diff
@@ -487,36 +460,89 @@ def parse_vcf(filteredFilePath, clumpsObjDict, tableObjDict, snpSet, clumpNumDic
+                                                index_snp_map[sample][clumpNum] = rsID, alleles if complements is None else complements
                                         else:
                                             # Since the study/name combo wasn't already used in the index map, add it to both the index and sample map
-                                            index_snp_map[sample][clumpNum] = rsID, alleles
+                                            index_snp_map[sample][clumpNum] = rsID, alleles if complements is None else complements
 
+                            #TODO: LOOK AT THIS
                             # the sample's alleles don't include the risk allele and early clumping is not requested
                             else:
-                                if sample not in viable_snp_counter:
-                                    no_viable_snp_counter.add(sample)
                                 unmatchedAlleleVariants.add(rsID)
 
                             clumped_snps_map[sample] = clumpedVariants
                             neutral_snps_map[sample] = unmatchedAlleleVariants
 
-        # Check to see which study/sample combos didn't have any viable snps
-        # and create blank entries for the sample map for those that didn't
-        for sample in no_viable_snp_counter:
-            sample_map[sample][""] = ""
+        # This next code accounts for snps that are in the study but are not reported in the sample. Instead of assuming the reference allele, we 
+        # assume that the allele is unknown and thus will use MAF for calculations of these snps
+        snpsLeftToImpute = set(snpSet).difference(usedSnps)
+        for sample in sampleOrder:
+            # Grab or create maps that hold sets of unused variants for this sample
+            clumpedVariants = clumped_snps_map[sample] if sample in clumped_snps_map else set()
+            for rsID in snpsLeftToImpute:
+                if trait in tableObjDict['associations'][rsID]['traits'] and study in tableObjDict['associations'][rsID]['traits'][trait] and pValBetaAnnoValType in tableObjDict['associations'][rsID]['traits'][trait][study]:
+                    pValue = tableObjDict['associations'][rsID]['traits'][trait][study][pValBetaAnnoValType]['pValue']
+                    # compare the pvalue to the pvalue cutoff
+                    if pValue <= float(p_cutOff):
+                        if rsID in clumpsObjDict:
+                            # Grab the clump number associated with this study and snp position
+                            clumpNum = str(clumpsObjDict[rsID]['clumpNum'])
+                            # Check to see how many variants are in this clump. If there's only one, we can skip the clumping checks.
+                            clumpNumTotal = clumpNumDict[clumpNum]
+
+                            if clumpNumTotal > 0: #TODO shouldn't this be > 1?
```

Comment:
is this TODO resolved?

Reply:
This is a question I have for @lizvance. Will you take a look at this section? Looking at the 
`# Check to see how many variants are in this clump. If there's only one, we can skip the clumping checks.
clumpNumTotal = clumpNumDict[clumpNum]`
it would seem to me that the number should be 1? but do we miss something if it is 1 and that is why it is zero?

Reply:
Oh good catch. Yeah, that should be a 1. Do you mind changing that as part of this branch?